internal/service/ocm/ocm_impl.go

```diff
@@ -0,0 +1,147 @@
+package ocm
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	sdk "github.com/openshift-online/ocm-sdk-go"
+	v1 "github.com/openshift-online/ocm-sdk-go/accountsmgmt/v1"
+	"github.com/openshift-online/ocm-sdk-go/logging"
+	"github.com/redhatinsights/mbop/internal/models"
+)
+
+type SDK struct {
+	client *sdk.Connection
+}
+
+func (ocm *SDK) InitSdkConnection(ctx context.Context) error {
+	// Create a logger that has the debug level enabled:
+	logger, err := logging.NewGoLoggerBuilder().
+		Debug(true).
+		Build()
+
+	if err != nil {
+		return err
+	}
+
+	ocm.client, err = sdk.NewConnectionBuilder().
+		Logger(logger).
+
+		// SA Auth:
+		Client(os.Getenv("COGNITO_APP_CLIENT_ID"), os.Getenv("COGNITO_APP_CLIENT_SECRET")).
+
+		// Offline Token Auth:
+		// Tokens(<token>).
+
+		// Oauth Token URL:
+		TokenURL(os.Getenv("OAUTH_TOKEN_URL")).
+
+		// Route to hit for AMS:
+		URL(os.Getenv("AMS_URL")).
+
+		// SA Scopes:
+		Scopes(os.Getenv("COGNITO_SCOPE")).
+		BuildContext(ctx)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ocm *SDK) GetUsers(usernames models.UserBody, q models.UserQuery) (models.Users, error) {
+	search := createSearchString(usernames)
+	collection := ocm.client.AccountsMgmt().V1().Accounts().List().Search(search)
+
+	collection = collection.Order(createQueryOrder(q))
+
+	users := models.Users{}
+	usersResponse, err := collection.Send()
+
+	if err != nil {
+		return users, err
+	}
+
+	if usersResponse.Items().Empty() {
+		return users, err
+	}
+
+	users = responseToUsers(usersResponse)
+
+	return users, err
+}
+
+func (ocm *SDK) IsOrgAdmin(id string) (bool, error) {
+	search := fmt.Sprintf("account.id='%s' and role.id='OrganizationAdmin'", id)
+
+	collection := ocm.client.AccountsMgmt().V1().RoleBindings()
+	roleBindings, err := collection.List().Search(search).Send()
+
+	if err != nil {
+		return false, err
+	}
+
+	if roleBindings.Items().Empty() {
+		return false, err
+	}
+
+	return true, err
```

Comment:
Not super opinionated on it, but what are your thoughts on flipping this so that we default to return `false`, unless we know they explicitly have a role binding for admin? Just thinking of potential edge-cases where it might pass above checks (really shouldn't with the `.Empty()` check) but if logic changes for instance, we might have an edge-case for a false positive?

Reply:
Will do!